### PR TITLE
Match (Most of) Firered

### PR DESCRIPTION
**Pull request by PokeCodec**
So this is pretty much all of the functions I was able to match. I am currently stuck on the others, as in Emerald, but now we are down to under 10 naked functions now in FireRed!

### CODE REVIEW

**Comment by PokeCodec** [comment]
This is all I could do.

File: asm/libagbsyscall.s [hunk 1]
```diff
@@ -7,62 +7,62 @@
 
 	thumb_func_start ArcTan2
 ArcTan2: @ 81E3B58
-	swi 0xA
+	svc 0xA
```
Comment (GriffinRichards) [inline asm/libagbsyscall.s:10]: Why are these instruction swaps included again

File: src/water.c [hunk 2]
```diff
@@ -874,14 +800,17 @@ static void AnimSmallBubblePair_Step(struct Sprite *sprite)
...
     u8 taskId2;
+    u16 *x, *y; //These pointers are needed to match
+
+    x = &gBattle_BG1_X;
+    y = &gBattle_BG1_Y;
 
-    SetGpuReg(REG_OFFSET_BLDCNT, BLDCNT_TGT1_BG1 | BLDCNT_EFFECT_BLEND | BLDCNT_TGT2_ALL);
-    SetGpuReg(REG_OFFSET_BLDALPHA, BLDALPHA_BLEND(0, 16));
+    SetGpuReg(REG_OFFSET_BLDCNT, BLDCNT_EFFECT_BLEND | BLDCNT_TGT2_ALL | BLDCNT_TGT1_BG1);
+    SetGpuReg(REG_OFFSET_BLDALPHA, (16 << 8));
```
Comment (GriffinRichards) [inline src/water.c:813]: These aren't necessary

File: src/water.c [hunk 3]
```diff
@@ -1230,167 +922,7 @@ static void AnimTask_CreateSurfWave_Step1(u8 taskId)
...
-	str r0, [r2]\n\
-_08107CA8:\n\
-	add sp, 0x10\n\
-	pop {r4-r7}\n\
-	pop {r0}\n\
-	bx r0\n\
-	.pool\n");
-}
-#endif
+
```
Comment (GriffinRichards) [inline src/water.c:925]: Extra newline

File: src/pokedex_screen.c [hunk 4]
```diff
@@ -977,9 +975,8 @@ void sub_8103AC8(u8 taskId)
             RemoveScrollIndicatorArrowPair(gUnknown_203ACF0->field_60);
             ListMenuRemoveCursorObject(gUnknown_203ACF0->field_61, 0);
             gUnknown_203ACF0->field_01 = 12;
+            break;
         }
-        else
-        {
             if (!JOY_HELD(R_BUTTON) && JOY_REPT(DPAD_LEFT))
```
Comment (GriffinRichards) [inline src/pokedex_screen.c:980]: Removed the else but didn't change the indentation

File: src/pokedex_screen.c [hunk 5]
```diff
@@ -242,21 +240,21 @@ void sub_810250C(void)
...
-        DecompressAndLoadBgGfxUsingHeap(3, gUnknown_8440274, BG_SCREEN_SIZE, 0, 0);
+        DecompressAndLoadBgGfxUsingHeap(3, (void*)gUnknown_8440274, BG_SCREEN_SIZE, 0, 0);
     InitWindows(gUnknown_8451ECC);
     DeactivateAllTextPrinters();
     m4aSoundVSyncOn();
     SetVBlankCallback(sub_81024C0);
     EnableInterrupts(INTR_FLAG_VBLANK);
     taskId = CreateTask(sub_810287C, 0);
-    gUnknown_203ACF0 = Alloc(sizeof(*gUnknown_203ACF0));
+    gUnknown_203ACF0 = (struct PokedexScreenData *)Alloc(sizeof(struct PokedexScreenData));
```
Comment (GriffinRichards) [inline src/pokedex_screen.c:257]: Why?

File: src/pokedex_screen.c [hunk 6]
```diff
@@ -75,34 +73,34 @@ struct PokedexScreenWindowGfx
...
-u16 sub_8103518(u8 a0);
-void sub_8103924(const struct ListMenuTemplate * a0, u8 a1);
-u8 sub_81039F0(void);
-void sub_8103988(u8 a0);
-void sub_8103AC8(u8 taskId);
-u8 sub_8104234(void);
-int sub_8104284(void);
-void sub_81042EC(u8 taskId);
-bool32 sub_8104664(u8 a0);
+EWRAM_DATA struct PokedexScreenData * gUnknown_203ACF0 = {0};
```
Comment (GriffinRichards) [inline src/pokedex_screen.c:76]: Why change this from `NULL`

File: src/pokedex_screen.c [hunk 7]
```diff
@@ -1384,274 +1380,98 @@ void sub_81042EC(u8 taskId)
...
-bool32 sub_8104664(u8 a0)
+static bool32 sub_8104664(u8 a0)
 {
-    u16 *r12;
-    u16 *r6;
     int r3;
+    u16	*r6, *r12;
 
-    switch (gUnknown_203ACF0->field_42)
+	switch (gUnknown_203ACF0->field_42)
```
Comment (GriffinRichards) [inline src/pokedex_screen.c:1388]: Converted spaces to tabs

File: src/menu.c [hunk 8]
```diff
@@ -559,7 +559,7 @@ static void CreateYesNoMenu2(const struct WindowTemplate *window, u8 paletteNum,
 
 s8 Menu_ProcessInputNoWrapClearOnChoose(void)
 {
-    s8 result = Menu_ProcessInputNoWrapAround();
+    const s8 result = Menu_ProcessInputNoWrapAround();
```
Comment (GriffinRichards) [inline src/menu.c:562]: `const` is unnecessary here

File: src/pokedex_screen.c [hunk 9]
```diff
@@ -242,21 +240,21 @@ void sub_810250C(void)
...
+    SetBgTilemapBuffer(3, (u16*)Alloc(BG_SCREEN_SIZE));
+    SetBgTilemapBuffer(2, (u16*)Alloc(BG_SCREEN_SIZE));
+    SetBgTilemapBuffer(1, (u16*)Alloc(BG_SCREEN_SIZE));
+    SetBgTilemapBuffer(0, (u16*)Alloc(BG_SCREEN_SIZE));
     if (natDex)
-        DecompressAndLoadBgGfxUsingHeap(3, gUnknown_84403AC, BG_SCREEN_SIZE, 0, 0);
+        DecompressAndLoadBgGfxUsingHeap(3, (void*)gUnknown_84403AC, BG_SCREEN_SIZE, 0, 0);
     else
-        DecompressAndLoadBgGfxUsingHeap(3, gUnknown_8440274, BG_SCREEN_SIZE, 0, 0);
+        DecompressAndLoadBgGfxUsingHeap(3, (void*)gUnknown_8440274, BG_SCREEN_SIZE, 0, 0);
```
Comment (GriffinRichards) [inline src/pokedex_screen.c:250]: Is there a reason for these casts?

File: src/water.c [hunk 10]
```diff
@@ -626,99 +626,25 @@ void AnimTask_RotateAuroraRingColors(u8 taskId)
...
-	adds r0, r4, 0\n\
-	bl DestroyAnimVisualTask\n\
-_081075DE:\n\
-	pop {r4-r7}\n\
-	pop {r0}\n\
-	bx r0\n\
-	.pool\n");
-}
-#endif
+
```
Comment (GriffinRichards) [inline src/water.c:647]: Extra newline

File: src/pokedex_screen.c (hunk 9 above)
Comment (PokeCodec) [reply src/pokedex_screen.c:250]: Yes, because otherwise gcc throws warnings

**Comment by PokeCodec** [comment]
Fixed!

**Comment by PokeCodec** [comment]
Done!

**Comment by PokeCodec** [comment]
Ready for review!

**Pull request merged**